zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/PersistedClusterTopologyTest.java

```diff
@@ -0,0 +1,107 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.qa.util.cluster.TestCluster;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import io.camunda.zeebe.test.util.junit.AutoCloseResources;
+import io.camunda.zeebe.test.util.junit.AutoCloseResources.AutoCloseResource;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import java.time.Duration;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+@ZeebeIntegration
+@AutoCloseResources
+@Timeout(2 * 60)
+public class PersistedClusterTopologyTest {
+
+  private static final int CLUSTER_SIZE = 3;
+  private static final int PARTITION_COUNT = 3;
+  private static final int REPLICATION_FACTOR = 3;
+
+  @AutoCloseResource private ZeebeClient client;
+
+  @TestZeebe
+  private final TestCluster cluster =
+      TestCluster.builder()
+          .useRecordingExporter(true)
+          .withGatewaysCount(1)
+          .withEmbeddedGateway(true)
+          .withBrokersCount(CLUSTER_SIZE)
+          .withPartitionsCount(PARTITION_COUNT)
+          .withReplicationFactor(REPLICATION_FACTOR)
+          .withGatewayConfig(
+              g ->
+                  g.gatewayConfig()
+                      .getCluster()
+                      .getMembership()
+                      // Decrease the timeouts for fast convergence of gateway topology. When the
+                      // broker is shutdown, the topology update takes at least 10 seconds with
+                      // the
+                      // default values.
+                      .setSyncInterval(Duration.ofSeconds(1))
+                      .setFailureTimeout(Duration.ofSeconds(5)))
+          .build();
+
+  @Test
+  void canHandleTopologyRequestsWhenBroker0IsRemoved() {
+    // given
+    cluster.awaitCompleteTopology();
+    client = cluster.newClientBuilder().build();
+
+    ///  Overview of the process being deployed
+    ///                      _____________________________
+    //     START_EVENT ->   |         service1           |     -------> END_EVENT
+    //                      |-----(message catch event)--|                 ^
+    //                                        |          _______________   |
+    //                                        |---------| serviceCatch | --|
+    //                                                  ---------------
+    final var deploymentEvent =
+        client
+            .newDeployResourceCommand()
+            .addResourceFromClasspath("processes/catch_event.bpmn")
+            .send()
+            .join();
+
+    cluster.shutdown();
+    RecordingExporter.reset();
+
+    // when
+    cluster.setPartitionCount(1);
+    cluster.start();
+    cluster.awaitCompleteTopology(
+        CLUSTER_SIZE, PARTITION_COUNT, REPLICATION_FACTOR, Duration.ofSeconds(10));
+
+    final var processId =
```

Comment:
I verified that without the fix the test does not pass with just one process instance, but I can increase the number of deployed processes just to be sure (at the cost of the tests taking more time)